modules/oe_webtools_maps/src/Plugin/Field/FieldFormatter/WebtoolsMapFormatter.php

```diff
@@ -64,17 +73,51 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
     $element = [];
 
     foreach ($items as $delta => $item) {
+      $data_array = [
+        'service' => 'map',
+        'version' => '2.0',
+        'map' => [
+          'zoom' => $this->getSetting('zoom_level'),
+          'center' => [$item->get('lat')->getValue(), $item->get('lon')->getValue()],
+        ],
+      ];
+
+      if ($this->getSetting('show_marker')) {
+        $data_array['layers'] = [
+          [
+            'markers' => [
+              'type' => 'FeatureCollection',
+              'features' => [
+                [
+                  'type' => 'Feature',
+                  'properties' => [
+                    'name' => $this->t('Coordinates'),
+                    'description' => $this->t('Longitude: @lon, Latitude: @lat', [
+                      '@lon' => $item->get('lon')->getValue(),
+                      '@lat' => $item->get('lat')->getValue(),
```

Comment:
I'm not sure if this is the best possible description but we have no other data available at this point. The data is coming from a geofield which only contains coordinates.

In real life this geofield will be populated by a text description and geocoded. So it would be great if we could retrieve the string that was used for geocoding, and show something relevant such as "Baker Street" instead of coordinates.

In other uses we would want to show the name of the building or organisation which is located at this place, rather than the street address.

I see that in the Joinup implementation this class is being overridden to provide a more meaningful description (ref. https://github.com/ec-europa/joinup-dev/pull/1872). I don't think that expecting that all users of this functionality will need to override the class is a good solution for this.

I am not sure at this point what would be the best way to solve this, maybe we could fire an event?

Reply:
I reconsider this. It is not so important since we can already use the existing `hook_ENTITY_TYPE_view()` to alter this trivially. It would be nice to maybe add some options to the field settings form to retrieve the name and description from the entity title or geocoder fields, but this can be added at a later stage.